### ISSUE TITLE
macOS build failed

### ISSUE THREAD

**Issue by erichrutz** [copperwater/xNetHack#104]
I have tried to execute the macOS build through sys/unix/install_macos.sh. I have an M1 MacBook with macOS 12.6.3 (Monterey).

The build runs fine until I run into this error:

[CC] monst.c
In file included from monst.c:6:
In file included from ../include/config.h:710:
In file included from ../include/integer.h:54:
In file included from /Library/Developer/CommandLineTools/usr/lib/clang/14.0.0/include/stdint.h:52:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/stdint.h:52:
In file included from /Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/_types.h:32:
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/sys/cdefs.h:309:39: **error: too few arguments provided to function-like macro invocation**
#if __has_attribute(warn_unused_result)
                                      ^
1 error generated.

Any advises on what I could do?

**Comment by RikerW** [comment]
For MacOS, try running

```
./sys/unix/setup.sh sys/unix/hints/macOS.370
make fetch-lua
make all
make install
``` 
Instead of that install file. The typical build path for unix is to run those commands, with the only thing being swapped out is the `sys/unix/hints/PATHNAME` hints file to actually run. Note: I don't use MacOS anymore, but I'm getting this off how I compile on ubuntu and the travis build workflow for this repo, [link](https://github.com/copperwater/xNetHack/actions/runs/4085904436/workflow).

**Comment by entrez** [comment]
This is a known issue that should be fixed with the next vanilla merge.  The vanilla fix is in commit NetHack/NetHack@13028143af8fbdc453ee261c957eac8617da9da4.  You can try applying that change yourself in the meantime, it's pretty straightforward.  You just add `&& !defined(MACOS)` so that `__warn_unused_result__` and `warn_unused_result` aren't "blanked out" in tradstdc.h:

```diff
diff --git a/include/tradstdc.h b/include/tradstdc.h
index 8c33d55e0..ee6bd99c2 100644
--- a/include/tradstdc.h
+++ b/include/tradstdc.h
@@ -405,7 +405,7 @@ typedef genericptr genericptr_t; /* (void *) or (char *) */
 #if __GNUC__ >= 3
 #define UNUSED __attribute__((unused))
 #define NORETURN __attribute__((noreturn))
-#if !defined(__linux__) || defined(GCC_URWARN)
+#if (!defined(__linux__) && !defined(MACOS)) || defined(GCC_URWARN)
 /* disable gcc's __attribute__((__warn_unused_result__)) since explicitly
    discarding the result by casting to (void) is not accepted as a 'use' */
 #define __warn_unused_result__ /*empty*/
```

**Comment by erichrutz** [comment]
Thank you very much. Applying the patch worked. Now I can compile
